tasks.py

```diff
@@ -221,29 +234,40 @@ def docker(ctx, name="gx38local", tag="latest", build=False, cmd="bash"):
             "The docker task must be invoked from the same directory as the task.py file at the top of the repo.",
             code=1,
         )
+
+    cmds = ["docker"]
+
     if build:
-        cmds = [
-            "docker",
-            "buildx",
-            "build",
-            "-f",
-            "docker/Dockerfile.tests",
-            f"--tag {name}:{tag}",
-            *[f"--build-arg {arg}" for arg in ["SOURCE=local", "PYTHON_VERSION=3.8"]],
-            ".",
-        ]
+        cmds.extend(
+            [
+                "buildx",
+                "build",
+                "-f",
+                "docker/Dockerfile.tests",
+                f"--tag {name}:{tag}",
+                *[
+                    f"--build-arg {arg}"
+                    for arg in ["SOURCE=local", f"PYTHON_VERSION={py}"]
+                ],
+                ".",
+            ]
+        )
 
     else:
-        cmds = [
-            "docker",
-            "run",
-            "-it",
-            "--rm",
-            "--mount",
-            f"type=bind,source={filedir},target=/great_expectations",
-            "-w",
-            "/great_expectations",
-            f"{name}:{tag}",
-            f"{cmd}",
-        ]
+        cmds.append("run")
+        if detach:
+            cmds.append("--detach")
```

Comment:
I have to insert the `--detach` here, or it doesn't work.
Or at least it doesn't work if I append to the end of the cmds.